Dockerfile

```diff
@@ -9,53 +17,87 @@ FROM python:3.7.11-slim-stretch
+# environment so that pipenv and its dependencies are not installed in the
+# Python virtual environment used in the final image.
+RUN python3 -m pip install --no-cache-dir --upgrade pipenv==${PYTHON_PIPENV_VERSION} \
+  # Manually create Python virtual environment for the final image
+  && python3 -m venv ${VIRTUAL_ENV} \
+  # Ensure the core Python packages are installed in the virtual environment
+  && ${VIRTUAL_ENV}/bin/python3 -m pip install --no-cache-dir --upgrade \
+    pip==${PYTHON_PIP_VERSION} \
+    setuptools==${PYTHON_SETUPTOOLS_VERSION} \
+    wheel==${PYTHON_WHEEL_VERSION}
+
+# Install code-gov-update Python requirements
+WORKDIR /tmp
+COPY src/Pipfile src/Pipfile.lock ./
+RUN pipenv sync --clear --verbose
+
+# The version of Python used here should match the version of the Alpine
+# python3 package installed in the compile-stage.
+FROM python:3.10.10-alpine3.17 as build-stage
+
+# Unprivileged user information
+ARG CISA_UID=2048
 ARG CISA_GID=${CISA_UID}
 ARG CISA_USER="cisa"
 ENV CISA_GROUP=${CISA_USER}
 ENV CISA_HOME="/home/${CISA_USER}"
+ENV VIRTUAL_ENV="${CISA_HOME}/.venv"
+
+# Install the dependencies needed by the llnl-scraper Python package to
+# estimate labor hours for code.
+RUN apk --no-cache add \
+  cloc=1.94-r0 \
+  git=2.38.4-r1
 
-###
 # Create unprivileged user
-###
-RUN groupadd --system --gid ${CISA_GID} ${CISA_GROUP} \
-    && useradd --system --uid ${CISA_UID} --gid ${CISA_GROUP} --comment "${CISA_USER} user" ${CISA_USER}
-
-##
-# Install cloc and git since llnl-scraper requires them to estimate
-# the labor hours.
-##
-RUN apt-get --quiet update \
-    && apt-get install --quiet --assume-yes \
-    cloc \
-    git
-
-##
-# Make sure pip and setuptools are the latest versions
-##
-RUN pip install --upgrade pip setuptools
-
-##
-# Install code-gov-update python requirements
-##
-COPY src/requirements.txt /tmp
-RUN pip install -r /tmp/requirements.txt
-
-# Clean up aptitude cruft
-RUN apt-get clean && rm -rf /var/lib/apt/lists/*
-
-# Put this just before we change users because the copy (and every
-# step after it) will often be rerun by docker, but we need to be root
-# for the chown command.
-COPY src/update.sh src/email-update.py src/body.txt src/body.html $CISA_HOME/
-RUN chown -R ${CISA_USER}:${CISA_USER} $CISA_HOME
+RUN addgroup --system --gid ${CISA_GID} ${CISA_GROUP} \
+  && adduser --system --uid ${CISA_UID} --ingroup ${CISA_GROUP} ${CISA_USER}
 
-###
-# Prepare to run
-###
+# Copy in the Python venv we created in the compile stage and re-symlink
+# python3 in the venv to the Python binary in this image
+COPY --from=compile-stage --chown=${CISA_USER}:${CISA_GROUP} ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+RUN ln -sf "$(command -v python3)" "${VIRTUAL_ENV}"/bin/python3
+ENV PATH="${VIRTUAL_ENV}/bin:$PATH"
+
+
+# Copy in the necessary files
+COPY --chown=${CISA_USER}:${CISA_GROUP} src/update.sh src/email-update.py src/body.txt src/body.html ${CISA_HOME}/
+
+# Prepare to Run
```

Comment:
Remove unnecessary capitalization:
```suggestion
# Prepare to run
```

Reply:
The curse of copying things from other projects sometimes~